doc/testing.md

```
@@ -0,0 +1,207 @@
+source .venv/bin/activate
+python -m pip install -e .
+```
+
+Optional clean start:
+
+```bash
+rm -rf .demo-single .demo-a .demo-b
+```
+
+
+## 3) Single-Device End-to-End 
```

markdownlint-cli2 (0.21.0)

[warning] Trailing spaces
Expected: 0 or 2; Actual: 1

(MD009, no-trailing-spaces)